plugins/in_kubernetes_events/kubernetes_events.c

```diff
@@ -679,6 +734,54 @@ static int k8s_events_sql_insert_event(struct k8s_events *ctx, msgpack_object *i
 
 #endif
 
+static int process_http_chunk(struct k8s_events* ctx, struct flb_http_client *c, 
+                size_t *bytes_consumed)
+{
+    int ret = 0;
+    char *token = NULL;
+    int root_type;
+    size_t consumed = 0;
+    char *buf_data = NULL;
+    size_t buf_size;
+    flb_sds_t payload;
+    
+    //we copy payload because tokenizer will overwrite when it finds end of string
+    payload = flb_sds_create_len(c->resp.payload, c->resp.payload_size);
+
+    token = strtok(payload, JSON_ARRAY_DELIM);
+    while ( token != NULL && ret == 0 ) {
+        ret = flb_pack_json(token, strlen(token), &buf_data, &buf_size, &root_type, &consumed);
+        if (ret == -1) {
+            flb_plg_error(ctx->ins, "could not process payload, incomplete or bad formed JSON: %s", token);
+        } else {
+            *bytes_consumed += strlen(token) + 1;
+            ret = process_watched_event(ctx, buf_data, buf_size);
+        }
+    
+        flb_free(buf_data);
+        if (buf_data) {
+            buf_data = NULL;
+        }
+        token = strtok(NULL, JSON_ARRAY_DELIM);
```

Comment:
we don't use strtok(). what do we need specifically here ?

Reply:
This function is processing data from the incoming http chunked stream (which is managed in the client.resp.payload) and I'm first creating a copy of that payload data into a string and then I use strtok is just to see if we have the `JSON_ARRAY_DELIM` available within any of the current data. If we do it makes a char* `token` to that string, then we attempt to parse it with the json lib. Then we just throw away the copied string. 

I can probably skip doing the copy and strtok and just use `strsep` on the client.resp.payload itself.

Reply:
good catch on this. I was not actually doing a second token check here... but i used `strpbrk` in the end and didn't need to make a copy of the payload string at all.